source/architecture_batch.md

```diff
@@ -0,0 +1,116 @@
+
+
+
+#### 1. Motivation
+
+User wants to run workloads using some trigger (cron, webhook, cloud events) that make predictions on some dataset
+using ML Model
+
+This dataset could have relatively large size and using blocking web api does not fit this usecase perfectly
+
+So we should provide async API that provide way to specify 
+
+1. Where input data is located (object storage link: s3, gcs, azure, etc)
+2. Where model is located (object storage link: s3, gcs, azure, etc) OR maybe some specific ML Registry such as MLFlow registry
+3. Where output data should be uploaded (object storage link: s3, gcs, azure, etc)
+4. Triggers that allow to run this job
+
+#### 2. REST API
+
+New manifests (REST API entities) are introduced
+
+1. BatchPredictor
+
+```yaml
+ID: wpm-matching-3-7
+Kind: BatchPredictor
+Spec:
+  image: <link to docker image>
+  entrypoint: python
+  cmd: ["run.py", "forecast", "-m", "top_emp"]
+  inputConnection: input-data
+  outputConnection: output-data
+  modelConnection: model-data
+  # If data will be not available in the moment of launch the predictor 
+  # will wait some time (retries * delayBetweenRetries) before stop with exception
+  waitForInputData:
+    retries: 15  
+    delayBetweenRetries: 120 
+  triggers:
+    webhook:
+      enabled: true
```

Comment:
We plan, for now, to add only `webhook` triggers or, maybe, it worth creating a separate entity for triggers? To have the possibility update only trigger URI or type of trigger?